### DISCUSSION TITLE
Issues with file downlink

### DISCUSSION THREAD

**Discussion by lukeclements** [nasa/fprime#2358] [Q&A]
Hi, I have two deployments, a RPI and bare metal controller, and I am trying to transfer a file from the RPi through the bare metal controller to the GDS. However, I am running into some issues with the file downlink component. When attempting to downlink a file a timeout occurs during the downlink process. My RPi receives the file downlink command, finds the file and starts the downlink operation. However, it times out and nothing is appearing in the GDS downlink tab.

An observation I made is that when I send the downlink command, the green circle in the GDS transitions to a red X momentarilly. I believe this is due to the downlink component attempting to transmit the file through the serial port. It seems to overwhelm the bandwidth of the serial port, resulting in the appearance of the red X. 

Any ideas on what the issue could be?

**Comment by LeStarch** [comment]
Does the red X disappear afterward? If not, it probably means there was an assertion failure somewhere and FSW crashed.  If it does come back to the green circle, then it means you are overwhelming your serial port.   In that case, are you running the ComQueue component?

**Reply by lukeclements** [reply]
The red X disappears, and I get a time out error on the RPi side. Neither the microcontroller or the RPi FSW crashes. 

I was originally trying to downlink an image, but instead tried a small .txt file, but it still times out. 

Yes, on the microcontroller side I am running the ComQueue component.

**Reply by LeStarch** [reply]
It sounds like there is something causing the microcontroller to crash...because otherwise I'd expect the data stream to recover.  Do you get any file packets down? In the `logs` folder there would be a log file for the downlinked file created and if that isn't there then nothing came down.

**Reply by lukeclements** [reply]
Sorry, I missed this reply. When I check the logs folder, I do not see anything related to downlink. A thought I have had is what happens to file downlink if telemetry is also downlinking, could this be causing issues?
